src/common/dav/os_thread.h

```diff
@@ -0,0 +1,167 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright 2015-2020, Intel Corporation */
+/*
+ * Copyright (c) 2016, Microsoft Corporation. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of the copyright holder nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * os_thread.h -- os thread abstraction layer
+ */
+
+#ifndef DAV_THREAD_H
+#define DAV_THREAD_H 1
+
+#include <stdint.h>
+#include <time.h>
+
+typedef union {
+	long long align;
+	char padding[44]; /* linux: 40 windows: 44 */
+} os_mutex_t;
+
+typedef union {
+	long long align;
+	char padding[56]; /* linux: 56 windows: 13 */
+} os_rwlock_t;
+
+typedef union {
+	long long align;
+	char padding[48]; /* linux: 48 windows: 12 */
+} os_cond_t;
+
+typedef union {
+	long long align;
+	char padding[32]; /* linux: 8 windows: 32 */
+} os_thread_t;
+
+typedef union {
+	long long align;  /* linux: long windows: 8 FreeBSD: 12 */
+	char padding[16]; /* 16 to be safe */
+} os_once_t;
+
+#define OS_ONCE_INIT { .padding = {0} }
+
+typedef unsigned os_tls_key_t;
+
+typedef union {
+	long long align;
+	char padding[56];  /* linux: 56 windows: 8 */
+} os_semaphore_t;
+
+typedef union {
+	long long align;
+	char padding[56];  /* linux: 56 windows: 8 */
+} os_thread_attr_t;
+
+typedef union {
+	long long align;
+	char padding[512];
+} os_cpu_set_t;
+
+typedef volatile int os_spinlock_t;
```

Comment:
(style)  do not add new typedefs
(style)  Use of volatile is usually wrong: see Documentation/volatile-considered-harmful.txt